src/Service.php

```diff
@@ -7,179 +8,315 @@
 namespace ZF\Apigility\Documentation\Swagger;
 
 use ZF\Apigility\Documentation\Service as BaseService;
+use ZF\Apigility\Documentation\Operation;
+use ZF\Apigility\Documentation\Field;
+use ZF\Apigility\Documentation\Swagger\Model\ModelGenerator;
 
 class Service extends BaseService
 {
+
+    const DEFAULT_TYPE = 'string';
+    const ARRAY_TYPE = 'array';
+
     /**
      * @var BaseService
      */
     protected $service;
+    protected $modelGenerator;
 
     /**
      * @param BaseService $service
-     * @param string $baseUrl
      */
-    public function __construct(BaseService $service, $baseUrl)
+    public function __construct(BaseService $service)
     {
         $this->service = $service;
-        $this->baseUrl = $baseUrl;
+        $this->modelGenerator = new ModelGenerator();
     }
 
     /**
      * @return array
      */
     public function toArray()
     {
-        // localize service object for brevity
-        $service = $this->service;
+        return $this->cleanEmptyValues([
+            'tags' => $this->getTags(),
+            'paths' => $this->cleanEmptyValues($this->getPaths()),
+            'definitions' => $this->getDefinitions()
+        ]);
+    }
+
+    protected function getTags()
```

Comment:
We favor `private` visibility over `protected` whenever possible. `protected` is part of the public API, and is thus problematic if we want to later change the signature or remove the method.